src/xcode/ENA/ENA/Source/Scenes/CreateAntigenTestProfile/CreateAntigenTestProfileViewController.swift

```diff
@@ -49,15 +118,290 @@ class CreateAntigenTestProfileViewController: UIViewController, FooterViewHandli
+		// view
+		view.backgroundColor = .enaColor(for: .background)
+		// tableView
+		tableView.separatorStyle = .none
+		tableView.register(DescriptionCell.self, forCellReuseIdentifier: DescriptionCell.identifier)
+		tableView.register(InputCell.self, forCellReuseIdentifier: InputCell.identifier)
+		tableView.keyboardDismissMode = .interactive
+		// date
+		dateFormatter = DateFormatter()
+		dateFormatter.dateStyle = .medium
+		dateFormatter.timeStyle = .none
+		dateFormatter.locale = Locale(identifier: "de_DE")
+	}
+	
+	private func setupBindings() {
+		viewModel.$antigenTestProfile
+			.sink { [weak self] antigenTestProfile in
+				let isSaveButtonEnabled =
+						!(antigenTestProfile.firstName?.isEmpty ?? true) ||
+						!(antigenTestProfile.lastName?.isEmpty ?? true) ||
+						(antigenTestProfile.dateOfBirth != nil) ||
+						!(antigenTestProfile.addressLine?.isEmpty ?? true) ||
+						!(antigenTestProfile.zipCode?.isEmpty ?? true) ||
+						!(antigenTestProfile.city?.isEmpty ?? true) ||
+						!(antigenTestProfile.phoneNumber?.isEmpty ?? true) ||
+						!(antigenTestProfile.email?.isEmpty ?? true)
+				self?.footerView?.setEnabled(isSaveButtonEnabled, button: .primary)
+			}
+			.store(in: &cancellables)
+	}
+	
+	@objc
+	private func firstNameTextFieldDidChange(textField: UITextField) {
+		viewModel.antigenTestProfile.firstName = textField.text
+	}
+	
+	@objc
+	private func lastNameTextFieldDidChange(textField: UITextField) {
+		viewModel.antigenTestProfile.lastName = textField.text
+	}
+	
+	@objc
+	private func birthDateTextFieldDidChange(textField: UITextField) {
+		viewModel.antigenTestProfile.dateOfBirth = nil
+	}
+	
+	@objc
+	private func dateOfBirthDidChange(datePicker: UIDatePicker) {
+		viewModel.antigenTestProfile.dateOfBirth = datePicker.date
+		if let cell = tableView.cellForRow(at: IndexPath(row: 1, section: 0)) as? InputCell {
+			cell.textFields[2].text = dateFormatter.string(from: datePicker.date)
+		}
+	}
+	
+	@objc
+	private func addressLineTextFieldDidChange(textField: UITextField) {
+		viewModel.antigenTestProfile.addressLine = textField.text
+	}
+	
+	@objc
+	private func postalCodeTextFieldDidChange(textField: UITextField) {
+		viewModel.antigenTestProfile.zipCode = textField.text
+	}
+	
+	@objc
+	private func cityTextFieldDidChange(textField: UITextField) {
+		viewModel.antigenTestProfile.city = textField.text
+	}
+	
+	@objc
+	private func phoneNumberTextFieldDidChange(textField: UITextField) {
+		viewModel.antigenTestProfile.phoneNumber = textField.text
+	}
+	
+	@objc
+	private func emailAddressTextFieldDidChange(textField: UITextField) {
+		viewModel.antigenTestProfile.email = textField.text
+	}
+}
 
+private class DescriptionCell: UITableViewCell {
+	
+	static let identifier = "DescriptionCell"
+		
+	override init(style: UITableViewCell.CellStyle, reuseIdentifier: String?) {
+		super.init(style: style, reuseIdentifier: reuseIdentifier)
+		// self
```

Comment:
Comments like that doesn't add any value. Please add more context to the comment. Or remove it. Please also see the other comments.